onnxruntime/test/testdata/custom_op_openvino_wrapper_library/openvino_wrapper.cc

```diff
@@ -0,0 +1,236 @@
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT32:
+      return ov::element::u32;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_UINT64:
+      return ov::element::u64;
+    case ONNX_TENSOR_ELEMENT_DATA_TYPE_BFLOAT16:
+      return ov::element::bf16;
+    default:
+      return ov::element::undefined;
+  }
+}
+
+static bool AreShapesEqual(const std::vector<int64_t>& ort_shape, const ov::Shape& ov_shape) {
+  if (ort_shape.size() != ov_shape.size()) {
+    return false;
+  }
+
+  const size_t num_dims = ort_shape.size();
+
+  for (size_t i = 0; i < num_dims; ++i) {
+    if (static_cast<decltype(ov_shape[i])>(ort_shape[i]) != ov_shape[i]) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+static bool AreIOTypesEqual(Ort::ConstTypeInfo type_info, const ov::Output<ov::Node>& ov_node) {
+  Ort::ConstTensorTypeAndShapeInfo type_shape_info = type_info.GetTensorTypeAndShapeInfo();
+
+  // Check element type.
+  const ov::element::Type ort_elem_type = ConvertONNXToOVType(type_shape_info.GetElementType());
+  const ov::element::Type ov_elem_type = ov_node.get_element_type();
+  if (ort_elem_type != ov_elem_type) {
+    return false;
+  }
+
+  // Check shape.
+  const std::vector<int64_t> ort_shape = type_shape_info.GetShape();
+  const ov::Shape& ov_shape = ov_node.get_shape();
+  if (!AreShapesEqual(ort_shape, ov_shape)) {
+    return false;
+  }
+
+  return true;
+}
+
+static bool ValidateInputsAndOutputs(const Ort::ConstKernelInfo& kinfo, const ov::OutputVector& ov_inputs,
+                                     const ov::OutputVector& ov_outputs) {
+  const size_t num_inputs = kinfo.GetInputCount();
+  const size_t num_outputs = kinfo.GetOutputCount();
+
+  // Number of inputs and outputs must match.
+  if (ov_inputs.size() != num_inputs || ov_outputs.size() != num_outputs) {
+    return false;
+  }
+
+  // Check input names, shapes, and element types.
+  for (size_t i = 0; i < num_inputs; ++i) {
+    Ort::TypeInfo ort_type = kinfo.GetInputTypeInfo(i);
+    const auto& ov_input = ov_inputs[i];
+
+    if (kinfo.GetInputName(i) != ov_input.get_any_name()) {
+      return false;
+    }
+
+    if (!AreIOTypesEqual(ort_type.GetConst(), ov_input)) {
+      return false;
+    }
+  }
+
+  // Check output names, shapes, and element types.
+  for (size_t i = 0; i < num_outputs; ++i) {
+    Ort::TypeInfo ort_type = kinfo.GetOutputTypeInfo(i);
+    const auto& ov_output = ov_outputs[i];
+
+    if (kinfo.GetOutputName(i) != ov_output.get_any_name()) {
+      return false;
+    }
+
+    if (!AreIOTypesEqual(ort_type.GetConst(), ov_output)) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+KernelOpenVINO::KernelOpenVINO(const OrtApi& /* api*/, const OrtKernelInfo* info,
+                               const std::unordered_map<std::string, std::string>& session_configs) {
+  Ort::ConstKernelInfo kinfo(info);
+
+  // Extract OpenVINO .bin and .xml contents from node attributes.
+  this->weights_ = kinfo.GetAttribute<std::string>("BIN");
```

Comment:
>this->weights_ = kinfo.GetAttribute<std::string>("BIN"); [](http://example.com/codeflow?start=2&length=56)

It would be good to demonstrate attributes other than string. For example, weights may be an array of floats.
For that matter, a single weight may be a Tensor. It would be great to test get a Tensor attribute.

Reply:
I'll update this example to extract the weights as a tensor of uint8s, which is what they actually are.

Reply:
Added public API to get tensor attributes.